src/constructRoutes.js

```diff
@@ -224,14 +240,16 @@ function validateAndSanitize(routesConfig) {
   const pathname = inBrowser ? window.location.pathname : "/";
   const hashPrefix = routesConfig.mode === "hash" ? pathname + "#" : "";
 
-  validateArray(
-    "routesConfig.routes",
-    routesConfig.routes,
-    validateRoute,
-    hashPrefix + routesConfig.base
-  );
+  validateArray("routesConfig.routes", routesConfig.routes, validateRoute, {
+    parentPath: hashPrefix + routesConfig.base,
+    siblingActiveWhens: [],
```

Comment:
`siblingActiveWhens` is the key to implementing default routes. We implement the activeWhen of the default routes to be the inverse of all of its sibling active whens. Note that "sibling" is applied a bit loosely here - dom element siblings are traversed recursively for nested routes, which count as "sibling" routes.